api/distributedcontext/map_test.go

```diff
@@ -0,0 +1,155 @@
+// Copyright 2019, OpenTelemetry Authors
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package distributedcontext
+
+import (
+	"fmt"
+	"testing"
+
+	"go.opentelemetry.io/api/core"
+	"go.opentelemetry.io/api/key"
+)
+
+func TestMap(t *testing.T) {
+	for _, testcase := range []struct {
+		name    string
+		init    []int
+		value   MapUpdate
+		wantKVs []core.KeyValue
+	}{
+		{
+			name: "NewMap with MultiKV",
+			value: MapUpdate{MultiKV: []core.KeyValue{
+				key.New("key1").Int64(1),
+				key.New("key2").String("val2")},
+			},
+			init: []int{},
+			wantKVs: []core.KeyValue{
+				key.New("key1").Int64(1),
+				key.New("key2").String("val2"),
+			},
+		},
+		{
+			name:  "NewMap with SingleKV",
+			value: MapUpdate{SingleKV: key.New("key1").String("val1")},
+			init:  []int{},
+			wantKVs: []core.KeyValue{
+				key.New("key1").String("val1"),
+			},
+		},
+		{
+			name: "NewMap with MapUpdate",
+			value: MapUpdate{SingleKV: key.New("key1").Int64(3),
+				MultiKV: []core.KeyValue{
+					key.New("key1").String(""),
+					key.New("key2").String("val2")},
+			},
+			init: []int{},
+			wantKVs: []core.KeyValue{
+				key.New("key1").String(""),
+				key.New("key2").String("val2"),
+			},
+		},
+		{
+			name:    "NewMap with empty MapUpdate",
+			value:   MapUpdate{MultiKV: []core.KeyValue{}},
+			init:    []int{},
+			wantKVs: []core.KeyValue{},
+		},
+		{
+			name: "Map with MultiKV",
+			value: MapUpdate{MultiKV: []core.KeyValue{
+				key.New("key1").Int64(1),
+				key.New("key2").String("val2")},
+			},
+			init: []int{5},
+			wantKVs: []core.KeyValue{
+				key.New("key1").Int64(1),
+				key.New("key2").String("val2"),
+				key.New("key5").Int(5),
+			},
+		},
+		{
+			name:  "Map with SingleKV",
+			value: MapUpdate{SingleKV: key.New("key1").String("val1")},
+			init:  []int{5},
+			wantKVs: []core.KeyValue{
+				key.New("key1").String("val1"),
+				key.New("key5").Int(5),
+			},
+		},
+		{
+			name: "Map with MapUpdate",
+			value: MapUpdate{SingleKV: key.New("key1").Int64(3),
+				MultiKV: []core.KeyValue{
+					key.New("key1").String(""),
```

Comment:
nit: We have recently added syntax like `key.String("key1", "")` to simplify these expressions. Would you consider using this syntax?

Reply:
Sure, no problem.